src/dune_rpc_server/dune_rpc_server.ml

```diff
@@ -87,6 +87,11 @@ module Session = struct
       | Uninitialized -> assert false
       | Initialized s -> t.state <- Initialized { s with closed = true }
 
+    let closed t =
+      match t.state with
+      | Uninitialized -> true
```

Comment:
Why is `Uninitialized` "closed"?

Reply:
Was thinking that an uninitialized session shouldn't be sending any messages, but I guess it doesn't matter

Reply:
If you don't expect this function to be called on uninitialized sessions, then perhaps it should be a code error?

But to me, `Uninitialized` as always closed just seems confusing. We should either move the `closed` flag to this state as well or return `false` instead of `true`.

Reply:
I'll raise a code error for now. I'm pretty unsatisfied with the whole `Stage1` architecture in general, but not enough so to rewrite it.